### PR TITLE
chore: update generate patch test source to log failures

### PR DESCRIPTION
**Pull request by mabdinur**
This script now log failures

Sample output

```
munirabdinur@COMP-GXQF74VFC6 dd-trace-py % python scripts/contrib-patch-tests.py                                                  
Generating patch test for boto
Generating patch test for flask
Generating patch test for algoliasearch
Generating patch test for aiohttp
Generating patch test for pymysql
Generating patch test for dogpile_cache
Generating patch test for futures
Generating patch test for mako
Generating patch test for mariadb
Generating patch test for aiohttp_jinja2
Generating patch test for kombu
Generating patch test for rediscluster
Generating patch test for aiobotocore
Generating patch test for asyncpg
Generating patch test for redis
Generating patch test for aiomysql
Generating patch test for mysqldb
Generating patch test for jinja2
Generating patch test for gevent
Generating patch test for pyramid
Generating patch test for pynamodb
Generating patch test for httplib
FAILED to generate patch test for httplib, required_modules not found in ddtrace.contrib.httplib.__init__
Generating patch test for pymongo
Generating patch test for botocore
Generating patch test for requests
Generating patch test for molten
Generating patch test for tornado
Generating patch test for pylons
Generating patch test for httpx
Generating patch test for mongoengine
Generating patch test for consul
Generating patch test for aiopg
Generating patch test for cassandra
Generating patch test for graphql
Generating patch test for sqlite3
FAILED to generate patch test for sqlite3, required_modules not found in ddtrace.contrib.sqlite3.__init__
Generating patch test for pyodbc
Generating patch test for aioredis
Generating patch test for mysql
Generating patch test for pylibmc
Generating patch test for sanic
Generating patch test for yaaredis
Generating patch test for aredis
Generating patch test for psycopg
Generating patch test for pymemcache
Generating patch test for aws_lambda
FAILED to generate patch test for aws_lambda, required_modules not found in ddtrace.contrib.aws_lambda.__init__
Generating patch test for vertica
Generating patch test for falcon
Generating patch test for logging
Generating patch test for bottle
Generating patch test for elasticsearch
FAILED to generate patch test for elasticsearch, required_modules not found in ddtrace.contrib.elasticsearch.__init__
```
## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Author is aware of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer is aware of, and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

**Comment by P403n1x87** [comment]
🙌 Thanks for improving this script! Not that this really matters, but I'd say this should be `WARNING` rather than `FAILED`

File: scripts/contrib-patch-tests.py [hunk 1]
```diff
@@ -29,15 +29,22 @@ def generate_patch_test_source(contrib):
     # Parse the contrib module to get the ``required_modules`` list
     # DEV: Parsing the module is much lighter and safer than importing it
     module_ast = ast.parse((CONTRIB_PATH / contrib / "__init__.py").read_text())
+    # require modules must be defined as module level array
     required_modules_node = [
         _ for _ in module_ast.body if isinstance(_, ast.Assign) and _.targets[0].id == "required_modules"
     ]
     if not required_modules_node:
+        print(
+            f"FAILED to generate patch test for {contrib}, required_modules not found in ddtrace.contrib.{contrib}.__init__"
```
Comment (mabdinur) [inline scripts/contrib-patch-tests.py:38]:
```suggestion
            f"WARNING to generate patch test for {contrib}, required_modules not found in ddtrace.contrib.{contrib}.__init__"
```

File: scripts/contrib-patch-tests.py [hunk 2]
```diff
@@ -29,15 +29,22 @@ def generate_patch_test_source(contrib):
...
+        )
         return
 
     required_modules = [_.value for _ in required_modules_node[0].value.elts]
 
     if not required_modules:
+        print(
+            f"FAILED to generate patch test for {contrib}, failed to parse ddtrace.contrib.{contrib}.required_modules"
```
Comment (mabdinur) [inline scripts/contrib-patch-tests.py:46]:
```suggestion
            f"WARNING: failed to generate patch test for {contrib}, ddtrace.contrib.{contrib}.required_modules could not be parsed"
```

File: scripts/contrib-patch-tests.py [hunk 3]
```diff
@@ -29,15 +29,22 @@ def generate_patch_test_source(contrib):
     # Parse the contrib module to get the ``required_modules`` list
     # DEV: Parsing the module is much lighter and safer than importing it
     module_ast = ast.parse((CONTRIB_PATH / contrib / "__init__.py").read_text())
+    # require modules must be defined as module level array
     required_modules_node = [
         _ for _ in module_ast.body if isinstance(_, ast.Assign) and _.targets[0].id == "required_modules"
     ]
     if not required_modules_node:
+        print(
+            f"WARNING: failed to generate patch test for {contrib}, required_modules not found in ddtrace.contrib.{contrib}.__init__"
```
Comment (mabdinur) [inline scripts/contrib-patch-tests.py:38]:
```suggestion
            f"WARNING: failed to generate patch test for {contrib},"
            " required_modules not found in ddtrace.contrib.{contrib}.__init__"
```

File: scripts/contrib-patch-tests.py [hunk 4]
```diff
@@ -29,15 +29,22 @@ def generate_patch_test_source(contrib):
...
+        )
         return
 
     required_modules = [_.value for _ in required_modules_node[0].value.elts]
 
     if not required_modules:
+        print(
+            f"WARNING: failed to generate patch test for {contrib}, ddtrace.contrib.{contrib}.required_modules could not be parsed"
```
Comment (mabdinur) [inline scripts/contrib-patch-tests.py:46]:
```suggestion
            f"WARNING: failed to generate patch test for {contrib},"
            " ddtrace.contrib.{contrib}.required_modules could not be parsed"
```

**Pull request merged**